tests/unit_tests/test_ocsp.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+const assert = require('assert');
+const simple = require('simple-mock');
+const commonUtils = require('../../src/utils/CommonUtils');
+const store = new (require("../../src/services/BeameStoreV2"))();
+
+const local_fqdn = process.env.BEAME_TESTS_LOCAL_FQDN;
+if (!local_fqdn) {
+	console.error(`local fqdn is required`);
+	process.exit(1)
+}
+
+const cred = store.getCredential(local_fqdn);
+if (!cred) {
+	throw new Error(`Credential for ${local_fqdn} not found`);
+}
+
+function mockRetryFn() {
+	return simple.mock(commonUtils, "retry").callFn(async function (func, retries = 5) {
+		let error = "";
+		for (let i = 1; i <= retries; ++i) { // retry
+			try {
+				return await func();
+			} catch (e) {
+				error = e;
+				console.log(`Call failed with error '${error}'. Retry [${i}/${retries}]`);
+			}
+		}
+		throw error;
+	});
+}
+
+describe('Test ocsp check', function () {
+	this.timeout(10000);
+
+	beforeEach(() => process.env.BEAME_THROW_OCSP = "");
+	afterEach(() => simple.restore());
+
+	const runs = [
+		{desc: '[Without Proxy] ', external_ocsp_fqdn: "", function_name: "checkOcspStatusWithoutExternalOcsp" },
+		{desc: '[With Proxy] ', external_ocsp_fqdn: "iep9bs1p7cj3cmit.tl5h1ipgobrdqsj6.v1.p.beameio.net", function_name: "checkOcspStatusWithExternalOcsp"}
+	];
+	runs.forEach(function (run) {
+
+		it(run.desc + 'without forceCheck', async () => {
+			process.env.EXTERNAL_OCSP_FQDN = run.external_ocsp_fqdn;
```

Comment:
Can `process.env.EXTERNAL_OCSP_FQDN = run.external_ocsp_fqdn;` go into `beforeEach` ?

Reply:
unfortunately not. I tried that by having the beforeeach inside the run, but it was not changing the value inside it per run. I could invest more time trying to understand why and if we can go around that behaviour. Is it worth it?

Reply:
no